### PR TITLE
Add DateInput component

### PR DESCRIPTION
**Pull request by cozmo**
This pulls the `DateInput` component out of the dashboard and into the components repo. No styling changes from the dashboard.

### CODE REVIEW

**Comment by CarlNelson** [comment]
Sadly not accessible at all :(

See Dropbox video > https://www.dropbox.com/s/ycnji9uxg3k169o/NotAccessible.mov?dl=0

File: src/DateInput/DateInput.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    this.setState({
+      showPicker: false,
+    });
+  }
+
+  render() {
+    const val = this.props.value ? moment(this.props.value) : moment();
+
+    return (<FlexBox column className={classnames("DateInput", this.props.className)}>
+      <div className="DateInput--InfoRow">{this.props.label}</div>
```
Comment (CarlNelson) [inline src/DateInput/DateInput.jsx]: This whole component should wrap <input type="date">

Not spans.

File: src/DateInput/DateInput.jsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+          type="link"
+          size="small"
+          value="Clear"
+          onClick={() => this._onChange(undefined)}
+        />}
+        <Button
+          className="DateInput--EditButton"
+          type="link"
+          size="small"
+          value={this.props.value ? "Change" : "Set"}
```
Comment (CarlNelson) [inline src/DateInput/DateInput.jsx]: Use `fa fa-calendar` with alt text "Set / Change Date"

http://fontawesome.io/icon/calendar/

File: src/DateInput/DateInput.jsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  }
+
+  render() {
+    const val = this.props.value ? moment(this.props.value) : moment();
+
+    return (<FlexBox column className={classnames("DateInput", this.props.className)}>
+      <div className="DateInput--InfoRow">{this.props.label}</div>
+      <FlexBox className="DateInput--DateRow">
+        <span className="DateInput--Date">
+          {this.props.value ? val.format("M/D/YYYY") : "--"}
```
Comment (CarlNelson) [inline src/DateInput/DateInput.jsx]: Don't use `--` as an empty state. Probably will be fixed when you convert to using the correct input type.

**Comment by cozmo** [comment]
Going a different direction with this.

**Pull request closed without merging**